subsys/net/ip/icmpv4.c

```diff
@@ -509,7 +509,7 @@ int net_icmpv4_send_echo_request(struct net_if *iface,
 	struct net_pkt *pkt;
 
 	if (!iface->config.ip.ipv4) {
-		return -EINVAL;
+		return -ENETDOWN;
```

Comment:
Either that one or -ENETUNREACH perhaps? Since not having an ipv4 configured does not mean the interface is down.
@jukkar what do you think?

Reply:
If the `iface->config.ip.ipv4` is NULL, it normally means that user has not created enough IPv4 config structs. This is affected by `CONFIG_NET_IF_MAX_IPV4_COUNT` option which basically tells how many network interfaces there will be in the system that will have IPv4 enabled. So it depends on how one has configured the system and it might be ok if ipv4 is NULL here. It does not mean that network interface is down as @tbursztyka said.
Perhaps -ENOTSUP (Unsupported value) would be more appropriate here, the -ENETUNREACH (Network is unreachable) is perhaps also feasible.

Reply:
@jukkar When running `samples/net/dhcpv4_client`, `iface->config.ip.ipv4` stays NULL if you do not plug the Ethernet cable (i.e. interface is down) and, when you do `net ping`, you get "Invalid IP address" message; this was the motivation behind this PR (maybe this in itself is a bug? or at least not an intended behaviour?).

Given the above circumstance, I think `ENETUNREACH` would be more adequate than `ENOTSUP`.

@jukkar @tbursztyka What do you think?

Reply:
fine by me

Reply:
> or at least not an intended behaviour?

Yep, there seems to be some side effect where the IPv4 is NULL in this case.

-ENETUNREACH is ok, we can certainly change that later if really needed.